recipes/wineditline/all/conanfile.py

```diff
@@ -1,64 +1,62 @@
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.53.0"
 
 
 class WineditlineConan(ConanFile):
     name = "wineditline"
-    description = (
-        "A BSD-licensed EditLine API implementation for the native "
-        "Windows Console"
-    )
+    description = "A BSD-licensed EditLine API implementation for the native Windows Console"
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "http://mingweditline.sourceforge.net/"
     topics = ("readline", "editline", "windows")
-    license = "BSD-3-Clause"
-    generators = ("cmake",)
-    settings = ("os", "arch", "compiler", "build_type")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
     }
     default_options = {
         "shared": False,
     }
-    exports_sources = ("patches/*", "CMakeLists.txt")
+    provides = "editline"
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", dst=self.export_sources_folder, src=self.recipe_folder)
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
         if self.settings.os != "Windows":
-            message = "wineditline is supported only on Windows."
-            raise ConanInvalidConfiguration(message)
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+            raise ConanInvalidConfiguration("wineditline is supported only on Windows.")
 
     def source(self):
-        root = self._source_subfolder
-        get_args = self.conan_data["sources"][self.version]
-        tools.get(**get_args, destination=root, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
+    def build(self):
+        copy(self, "CMakeLists.txt", dst=self.source_folder, src=self.export_sources_folder)
         cmake = CMake(self)
         cmake.configure()
-        return cmake
-
-    def build(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
-        self._configure_cmake().build()
+        cmake.build()
 
     def package(self):
-        self.copy("COPYING", "licenses", self._source_subfolder)
-        self._configure_cmake().install()
+        copy(self, "COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
```

Comment:
If we want to use original CMake, the following additions are needed (install method doesn't populate package correctly for tests to work instead):

` copy(self, "*.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include", "editline"), keep_path=False)  `
`copy(self, "*.exe", src=self.build_folder, dst=os.path.join(self.package_folder, "bin"), keep_path=False)  `
`copy(self, "*.dll", src=self.build_folder, dst=os.path.join(self.package_folder, "lib"), keep_path=False)  `
`copy(self, "*.lib", src=self.build_folder, dst=os.path.join(self.package_folder, "lib"), keep_path=False)  `